app/utilities_endpoints.py

```diff
@@ -189,3 +196,97 @@ def find_the_gene(range=None):
         output.append(ord_dict)
 
     return (jsonify(output))
+
+
+def translate_terminology(codeSystem, code):
+    # validate input parameters
+    if codeSystem not in ['http://www.nlm.nih.gov/research/umls/rxnorm', 'http://snomed.info/sct', 'http://hl7.org/fhir/sid/icd-10']:
+        return "unrecognized code system"
+    response = [{}]
+
+    # process rxnorm input
+    if codeSystem == 'http://www.nlm.nih.gov/research/umls/rxnorm':
+        MapRxNorm = fetch_concept_map(44872525)
+        if MapRxNorm.status_code == 200:
+            MapRxNorm = MapRxNorm.json()
+            for element in MapRxNorm["group"][0]["element"]:
+                if element["code"] == code:
+                    response[0]["outcome"] = 'match found'
+                    response[0]["code"] = element["target"][0]["code"]
+                    response[0]["system"] = 'https://ncithesaurus.nci.nih.gov/'
+                    response[0]["display"] = element["target"][0]["display"]
+                    return response
+            response[0]["outcome"] = 'no match found'
+            response[0]["system"] = 'https://ncithesaurus.nci.nih.gov/'
+            return response
+        else:
+            return "server down"
```

Comment:
Should we return HTTP 500 error in such cases? Something like `abort(500, "HAPI server error")`

Reply:
great suggestion, thank you.